ztp/ztp-policy-generator/kustomize/plugin/policyGenerator/v1/policygenerator/policyGenerator_test.go

```diff
@@ -0,0 +1,362 @@
+const OBJECT_DEFINITION="objectDefinition"
+
+/* Section Test Setup Functions Starts */
+func testSetup(t *testing.T) {
+	os.RemoveAll(getOutPath(t))
+}
+
+func testCleanup(t *testing.T) {
+	os.RemoveAll(getOutPath(t))
+}
+/* Section Test Setup Functions Ends */
+
+/* Section Test Functions Starts */
+
+/*func TestExample(t *testing.T) {
+	testSetup(t)
+	generateACMResourceDefinitions(t)
+	ACMResourceDefinitionAssertions(t)
+	generateCustomResourceDefinitions(t)
+	CustomResourceDefinitionAssertions(t)
+	testCleanup(t)
+}*/
+
+func TestNamespace(t *testing.T) {
+	testSetup(t)
+	generateACMResourceDefinitions(t)
+	ACMResourceDefinitionAssertions(t)
+	generateCustomResourceDefinitions(t)
+	CustomResourceDefinitionAssertions(t)
+	testCleanup(t)
+}
+
+func TestOperatorGroup(t *testing.T) {
+	testSetup(t)
+	generateACMResourceDefinitions(t)
+	ACMResourceDefinitionAssertions(t)
+	generateCustomResourceDefinitions(t)
+	CustomResourceDefinitionAssertions(t)
+	testCleanup(t)
+}
+
+func TestSubscription(t *testing.T) {
+	testSetup(t)
+	generateACMResourceDefinitions(t)
+	ACMResourceDefinitionAssertions(t)
+	generateCustomResourceDefinitions(t)
+	CustomResourceDefinitionAssertions(t)
+	testCleanup(t)
+}
+
+
+func TestMachineConfigPool(t *testing.T) {
+	testSetup(t)
+	generateACMResourceDefinitions(t)
+	ACMResourceDefinitionAssertions(t)
+	generateCustomResourceDefinitions(t)
+	CustomResourceDefinitionAssertions(t)
+	testCleanup(t)
+}
+
+func TestSriovNetwork(t *testing.T) {
+	testSetup(t)
+	generateACMResourceDefinitions(t)
+	ACMResourceDefinitionAssertions(t)
+	generateCustomResourceDefinitions(t)
+	CustomResourceDefinitionAssertions(t)
+	testCleanup(t)
+}
+
+func TestPtpConfig(t *testing.T) {
+	testSetup(t)
+	generateACMResourceDefinitions(t)
+	ACMResourceDefinitionAssertions(t)
+	generateCustomResourceDefinitions(t)
+	CustomResourceDefinitionAssertions(t)
+	testCleanup(t)
+}
+/* Section Test Functions Ends */
+
+/* Section Test Trigger Functions Starts */
+func generateACMResourceDefinitions(t *testing.T) {
+	InitiatePolicyGen (getTemplatePath(t), getSourcePolicyPath(t),getOutPath(t), true,false)
+}
+
+func generateCustomResourceDefinitions(t *testing.T) {
+	InitiatePolicyGen (getTemplatePath(t), getSourcePolicyPath(t),getOutPath(t), true,true)
+}
+/* Section Test Trigger Functions Ends */
+
+
+/* Section Assertions Functions Starts */
+func ACMResourceDefinitionAssertions(t *testing.T) {
+	dirName,fileName := computeACMCustomResourceDirectoryNameAndFileName(t)
+	dir,file := path.Split(dirName)
+	parentDir := getOutPath(t)
+	if len(dir) > 0 {
+		parentDir = path.Join(getOutPath(t), dir)
+		dirName = file
+	}
+	assert.True(t,checkDirectoryExists(parentDir, dirName))
+	var dirPath = path.Join(parentDir, dirName)
+	assert.True(t, checkFileExists(dirPath, fileName))
+	placementRuleAndBindingAssertions(dirPath, t)
+	var ACMResourceFilePath = path.Join(dirPath,fileName)
+	checkIfSourceYamlEqualsACMObjectDefinition(ACMResourceFilePath, t)
+
```

Comment:
This line is extra.

Reply:
We should always run our go code through `gofmt` to fixup the formatting.  In fact, this is now enforced on all GitHub PRs :)

Reply:
updated after running `gofmt`